token/program-2022/src/extension/mod.rs

```diff
@@ -353,6 +353,37 @@ fn get_extension_bytes_mut<S: BaseState, V: Extension>(
     Ok(&mut tlv_data[value_start..value_end])
 }
 
+/// Calculate the new expected size if the state allocates the given number
+/// of bytes for the given extension type.
+///
+/// Provides the correct answer regardless if the extension is already present
+/// in the TLV data.
+fn try_get_new_account_len_for_extension_len<S: BaseState, V: Extension>(
+    tlv_data: &[u8],
+    new_extension_len: usize,
+) -> Result<usize, ProgramError> {
+    // get the new length used by the extension
+    let new_extension_tlv_len = add_type_and_length_to_len(new_extension_len);
+    let tlv_info = get_tlv_data_info(tlv_data)?;
+    // If we're adding an extension, then we must have at least BASE_ACCOUNT_LENGTH
+    // and account type
+    let current_len = tlv_info
+        .used_len
+        .saturating_add(BASE_ACCOUNT_AND_TYPE_LENGTH);
+    let new_len = if tlv_info.extension_types.is_empty() {
+        current_len.saturating_add(new_extension_tlv_len)
```

Comment:
I did some testing around, and this branch could be safely removed without affecting the correctness of the code. How about we just remove it?

Reply:
That's weird I plucked this from the original. Removed.